src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -25,33 +27,40 @@ public class ScanResultToComponentListTransformer {
      * @param rapidScanFullResults
      * @return list of {@link Component}s
      */
-    public List<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
+    public Set<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
         HashMap<String, ScanMetadata> componentIdWithMetadata = new HashMap<>();
-
+        Set<String> componentNamesWithNullIds = new HashSet<>(rapidScanFullResults.size());
         for (DeveloperScansScanView component : rapidScanFullResults) {
-            componentIdWithMetadata.put(component.getExternalId(), populateMetadata(component));
+            if (component.getExternalId() == null) {
+                componentNamesWithNullIds.add(component.getComponentName());
+            } else {
+                componentIdWithMetadata.put(component.getExternalId(), populateMetadata(component));
+            }
+        }
+        for (String componentNameToWarnAbout : componentNamesWithNullIds) {
+            logger.warn("Component '{}' is skipped due to missing external ID in the scan result json.", componentNameToWarnAbout);
```

Comment:
To print the names once each instead of N number of times since they can and do repeat.